sbi/inference/snpe/snpe_base.py

```diff
@@ -155,10 +155,16 @@ def __call__(
 
             # Run simulations for the round.
             theta, x, prior_mask = self._run_simulations(round_, num_sims)
+            # XXX: Bypassing binary calibration kernel here. We need this to get rid of
+            # NaN simulations. We could do that using calibration kernels, but then we
+            # hack here can be added to sre and snl easily.
+
+            # Get indices with finite data to remove NaN simulations.
+            x_not_nan = torch.unique(torch.where(torch.isfinite(x))[0])
```

Comment:
Continuing on the theme above: an expressively-named function would get rid here of the need for a comment.